salt/states/boto_ec2.py

```diff
@@ -1752,5 +1749,169 @@ def private_ips_absent(name, network_interface_name=None, network_interface_id=N
 
         # there were no changes since we did not attempt to remove ips
         ret['changes'] = {}
+    return ret
+
+
+def secgroup_present(
```

Comment:
State modules are idempotent, so usually, rather than having present/equal/absent states for something you're going to apply to a resource, you'd add security groups as an attribute of the instance itself, and not manage attributes of a resource through several unrelated states.

Are you adding these because you have instances that are not managed by salt, but you want to manage parts of their resources? I'm not necessarily opposed to these states, since they're idempotent, but they're a bit out of consistency with how other things are done.

Does the instance_present module have a security groups attribute? If not, it would be ideal for this support to also be added there.